src/libServer/ScillaIPCServer.cpp

```diff
@@ -15,64 +15,78 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-#include "ScillaIPCServer.h"
+#include <jsonrpccpp/server/connectors/unixdomainsocketserver.h>
+
 #include "libPersistence/ContractStorage2.h"
 #include "libUtils/DataConversion.h"
 
-using namespace jsonrpc;
+#include "ScillaIPCServer.h"
+
 using namespace std;
 using namespace Contract;
 
-ScillaIPCServer::ScillaIPCServer(UnixDomainSocketServer &server,
-                                 const dev::h160 &contract_address)
-    : AbstractServer<ScillaIPCServer>(server, JSONRPC_SERVER_V2) {
-  this->bindAndAddMethod(Procedure("fetchStateValue", PARAMS_BY_NAME,
-                                   JSON_STRING, "query", JSON_STRING, NULL),
-                         &ScillaIPCServer::fetchStateValueI);
-
-  this->bindAndAddMethod(
-      Procedure("updateStateValue", PARAMS_BY_NAME, JSON_BOOLEAN, "query",
-                JSON_STRING, "value", JSON_STRING, NULL),
+ScillaIPCServer::ScillaIPCServer(const dev::h160 &contrAddr,
+                                 jsonrpc::UnixDomainSocketServer &conn,
+                                 jsonrpc::serverVersion_t type)
+    : jsonrpc::AbstractServer<ScillaIPCServer>(conn, type),
```

Comment:
Also, I just realized this, but in the client code we explicitly say the protocol being used is jsonrpc v2. Since we're adding the ability to use a different version here, should we perhaps also change the client? (or alternatively mandate that its v2)

Reply:
Thank you @advaypal . Good catch. I've mandated V2 now.